src/seeds.proposals.cpp

```diff
@@ -212,6 +212,28 @@ void proposals::update(uint64_t id, string title, string summary, string descrip
   });
 }
 
+void proposals::cancel(uint64_t id) {
+  auto pitr = props.find(id);
+
+  check(pitr != props.end(), "Proposal not found");
+  require_auth(pitr->creator);
+  check(pitr->status == name("open"), "Proposal state is not open, it can no longer be cancelled");
+
+  props.modify(pitr, _self, [&](auto & proposal) {
+    proposal.status = name("cancel");
+  });
+}
+
+void proposals::refund(uint64_t id) {
+  auto pitr = props.find(id);
+
+  check(pitr != props.end(), "Proposal not found");
+  require_auth(pitr->creator);
+  check(pitr->status == name("cancel"), "Proposal state must be cancel");
+
+  withdraw(pitr->creator, pitr->staked, contracts::bank);
```

Comment:
This needs to delete the proposal from the table 

Otherwise I can call cancel multiple times, and receive a refund multiple times.

Alternatively, set "staked" to 0.

Reply:
Ok I am implementing this now, I'll delete from table